cmd/acra-keys/keys/destroy-key.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020, Cossack Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package keys
+
+import (
+	"github.com/cossacklabs/acra/keystore"
+	log "github.com/sirupsen/logrus"
+)
+
+// SupportedDestroyKeyKinds is a list of keys supported by `destroy-key` subcommand.
+var SupportedDestroyKeyKinds = []string{
+	KeyTransportConnector,
+	KeyTransportServer,
+	KeyTransportTranslator,
+}
+
+// DestroyKey destroys data of the requsted key.
+func DestroyKey(params *CommandLineParams, keyStore keystore.KeyMaking) error {
+	switch params.DestroyKeyKind {
+	case KeyTransportConnector:
+		if params.ClientID == "" {
```

Comment:
why not to move this check out of switch and validate with function like `func validateClientID(keyKind <type, clientID []byte) bool`? Each check different from each other by name of kind of key which implement `String()` method and correctly formatted by .Errorf.

Reply:
> why not to move this check out of switch and validate with function

I'm not really sure it's worth it. Each key kind has its own requirements. Some need client ID, some need zone ID, some don't need anything. I've placed the checks into individual `case` clauses to keep them near the code that uses client/zone ID so it's more obvious why we check for that.

Though, it in this case we mostly do this to check that the user has specified `--client_id` on the command line when it's required by the key kind. So it makes sense to move this to command-line validation. Though, we'd need to parse the subcommand in order to know what key kinds we care about. So it will be more natural to have some

```go
func (params *CommandLineParams) CheckForKeyKind(keyKind string) error {
    // if something required is not set on the command line, return an error
}
```

How does that sound?

It does not seem to be very reusable outside of acra-keys (which is the only tool that cares about key kinds directly), but it should reduce duplication of such checks when more subcommands are added.